aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -162,7 +173,28 @@ private String cleanWarehousePath(String path) {
 
   @Override
   protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
-    return new GlueTableOperations(glue, lockManager, catalogName, awsProperties, fileIO, tableIdentifier);
+    if (isS3FileIOConfigured()) {
+      // Create different S3FileIO to use different credentials in future
+      FileIO s3FileIO = new S3FileIO(toTags(tableIdentifier));
+      s3FileIO.initialize(this.catalogProperties);
```

Comment:
Nit: I think the practice in ICeberg is to only use this when actually mutating the field, not needed when reading.

Reply:
Thanks, I have made the changes.